src/util/Tokenizer.hpp

```diff
@@ -32,9 +32,10 @@ class Tokenizer
 {
 public:
   enum class Mode {
-    include_empty,   // Include empty tokens.
-    skip_empty,      // Skip empty tokens.
-    skip_last_empty, // Include empty tokens except the last one.
+    include_empty,     // Include empty tokens.
+    skip_empty,        // Skip empty tokens.
+    skip_last_empty,   // Include empty tokens except the last one.
+    include_delimiter, // Include the delimiter
```

Comment:
I like the idea of being able not to strip the delimiter, but I'm not so sure about adding it to the `Mode` list. `Mode` says which tokens to include, while `include_delimiter` says how to process each token, so it's an odd bird. In theory one might want to retain delimiters but also skip empty tokens.

How about adding a fourth `bool strip_delimiter = true` or `bool include_delimiter = false` parameter to `Tokenizer::Tokenizer` instead? (Or even better: an enum class with two values.) It would be OK to `ASSERT(strip_delimiter || mode != include_empty);` for now if that simplifies the implementation (should be essentially the same as what you have now).

Reply:
Ok